miio/chuangmi_camera.py

```diff
@@ -375,13 +379,29 @@ def set_nas_config(
         video_retention_time: NASVideoRetentionTime = NASVideoRetentionTime.Week,
     ):
         """Set NAS configuration."""
-        if share is None:
-            share = {}
-        return self.send(
-            "nas_set_config",
-            {
-                "state": state,
-                "sync_interval": sync_interval,
-                "video_retention_time": video_retention_time,
-            },
-        )
+        share_obj = None
+        share = urlparse(share)
+        if share.scheme == "smb":
+            ip = socket.gethostbyname(share.hostname)
+            reversed_ip = ".".join(reversed(ip.split(".")))
+            addr = int(ipaddress.ip_address(reversed_ip))
+
+            share_obj = {
+                "type": 1,
+                "name": share.hostname,
+                "addr": addr,
+                "dir": share.path.lstrip('/'),
+                "group": "WORKGROUP",
+                "user": share.username,
+                "pass": share.password,
+            }
+        
+        params = {
+            "state": state,
+            "sync_interval": sync_interval,
+            "video_retention_time": video_retention_time,
+        }
+        if share_obj != None:
+            params["share"] = share_obj
```

Comment:
It would be better to move the `params` definition above the urlparse check, this way the smb code path just creates the share object and sets it to the already existing params dictionary.

Reply:
OK